parallel/runner.go

```diff
@@ -263,20 +216,15 @@ func (r *runner) addThread() {
 					break
 				}
 			}
-			r.idle.Store(threadId, true)
-			// Save the current time as the thread's last active time.
-			r.lastActive.Store(threadId, strconv.FormatInt(time.Now().Unix(), 10))
 
-			r.runningThreadsLock.Lock()
+			r.openThreadsLock.Lock()
 			// If there are more open threads than maxParallel, then this thread will be closed.
```

Comment:
// If there are more open threads than maxParallel, then this thread will be closed.
-->
// If there the total of open threads is larger than the maximim (maxParallel), then this thread should be closed.

Reply:
done